File: 73-set-matrix-zeroes/73-set-matrix-zeroes.py
class Solution(object):
    def setZeroes(self, mat):
        """
        :type matrix: List[List[int]]
        :rtype: None Do not return anything, modify matrix in-place instead.
        """
        n = len(mat)
        m = len(mat[0])
        r ,c = set(),set()
        for i in range(n):
            for j in range(m):
                if mat[i][j] == 0:
                    c.add(j)
                    r.add(i)
        
        for i in range(n):
            for j in range(m):
                if i in r or j in c:
                    mat[i][j] = 0

File: 73-set-matrix-zeroes/73-set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& mat) {
       set<pair<int,int>>s;
    for(int i = 0;i<mat.size();i++)
    {
        for(int j  = 0;j<mat[0].size();j++)
        {
            if(mat[i][j] == 0)
            {
                s.insert({i,j});
            }
        }
    }
    int n = mat.size();
    int m = mat[0].size();
    
    for(auto it : s)
    {
        int r = it.first;
        int c = it.second;
        for(int i  = 0;i<m;i++)
        {
            mat[r][i] = 0;
            
        }
        for(int i = 0;i<n;i++)
        {
            mat[i][c] = 0;
        }
    }
        
    }
};